src/pubsub/mod_pubsub_db_rdbms_sql.erl

```diff
@@ -420,80 +103,15 @@ common_node_insert(EscKey, EscName, EscType, EscOwners, EscOptions) ->
-             " WHERE nidx = ", EscNidx, ";"],
-    {updated, _} = mongoose_rdbms:sql_query(global, Query).
-
 set_parents(Node, Parents) ->
     EscNode = esc_string(Node),
     ParentRows = [parent_row(EscNode, Parent) || Parent <- Parents],
     ["INSERT INTO pubsub_node_collections (name, parent_name) "
      "VALUES ", rdbms_queries:join(ParentRows, ",")].
 
-del_parents(Node) ->
-    ["DELETE FROM pubsub_node_collections ",
-     "WHERE name = ", esc_string(Node)].
-
 parent_row(EscNode, Parent) ->
     ["(", EscNode, ", ", esc_string(Parent),")"].
 
-select_node_by_key_and_name(Key, Name) ->
-    ["SELECT ", pubsub_node_fields(), " from pubsub_nodes "
-     "WHERE p_key = ", esc_string(Key),
-     " AND name = ", esc_string(Name)].
-
--spec select_node_by_id(mod_pubsub:nodeIdx()) -> iolist().
-select_node_by_id(Nidx) ->
-    ["SELECT ", pubsub_node_fields(), " from pubsub_nodes "
-     "WHERE nidx = ", esc_int(Nidx)].
-
--spec select_nodes_by_key(Key :: binary()) -> iolist().
-select_nodes_by_key(Key) ->
-    ["SELECT ", pubsub_node_fields(), " from pubsub_nodes "
-     "WHERE p_key = ", esc_string(Key)].
-
--spec select_nodes_by_owner(LJID :: binary()) -> iolist().
-select_nodes_by_owner(LJID) ->
-    %% TODO I wrote that code in tears in my eyes. Its super inefficient,
-    %% there should be separate table for many-to-many relation and index
-    case {mongoose_rdbms:db_engine(global), mongoose_rdbms_type:get()} of
-        {mysql, _} ->
-            ["SELECT name, type"
-                " FROM pubsub_nodes"
-                " WHERE owners = convert(", esc_string(iolist_to_binary(["[\"", LJID, "\"]"])), ", JSON);"
-            ];
-        {pgsql, _}  ->
-            ["SELECT name, type"
-            " FROM pubsub_nodes"
-            " WHERE owners ::json->>0 like ", esc_string(LJID),
-                " AND JSON_ARRAY_LENGTH(owners) = 1"
-            ];
-        {odbc, mssql} ->
-            ["SELECT name, type"
-            " FROM pubsub_nodes"
-            " WHERE cast(owners as nvarchar(max)) = ", esc_string(iolist_to_binary(["[\"", LJID, "\"]"]))
-            ]
-    end.
-
--spec select_nodes_by_affiliated_user(LU :: jid:luser(), LS :: jid:lserver()) -> iolist().
-select_nodes_by_affiliated_user(LU, LS) ->
-    ["SELECT aff, ", pubsub_node_fields("pn"),
-     " FROM pubsub_affiliations AS pa"
-     " INNER JOIN pubsub_nodes AS pn ON pa.nidx = pn.nidx"
-     " WHERE luser = ", esc_string(LU),
-       " AND lserver = ", esc_string(LS)].
-
--spec select_nodes_in_list_with_key(Key :: binary(), Nodes :: [binary()]) -> iolist().
-select_nodes_in_list_with_key(Key, Nodes) ->
-    EscapedNames = [esc_string(Node) || Node <- Nodes],
-    NodeNames = rdbms_queries:join(EscapedNames, ","),
-    ["SELECT ", pubsub_node_fields(), " from pubsub_nodes "
-     "WHERE p_key = ", esc_string(Key),
-     " AND name IN (", NodeNames, ")"].
-
 -spec select_nodes_by_key_and_names_in_list_with_parents(Key :: binary(), Nodes :: [binary()]) -> iolist().
```

Comment:
These two are quite tricky. Currently I would just do one query per each `pn.name` and concatenate the results later. We could optimize it separately if we find a need for it.